include/bronto/bronto.hpp

```diff
@@ -79,6 +104,74 @@
 #define BRONTO_AFTER()
 #endif
 
+// `BRONTO_USAGE(tag)`:
+//
+// A function-like macro that can be applied to a `bronto::rewrite_*`
+// struct/class nested inside a struct/class inheriting from
+// `bronto::rewrite_decl`. Indicates that in order for the rewrite to be applied
+// to that declaration, some requirements are placed on the uses of the declared
+// variable. Specifically,
+//
+// * `BRONTO_USAGE(required)`: Requires that at least one use of the declared
+//   variable satisfies the constraints specified by the rule for the rewrite to
+//   occur. If the rewrite does occur, any matching usage will be rewritten
+//   according to this rule.
+//
+// * `BRONTO_USAGE(allowed)`: Makes no requirements about the usage of the
+//   declared variable, but rewrites each usage that satisfies the constraints
+//   of the rule.
+//
+// * `BRONTO_USAGE(forbidden)`: Requires that no use of the declared variable
+//   may satisfiy the constraints specified by the rule for the rewrite to
+//   occur.
+//
+// If multiple rules have a `BRONTO_USAGE(tag)` attribute, all constraints are
+// required. See https://brontosource.dev/docs for more details.
```

Comment:
I would like to have this link point to a specific page in the docs, but that is a TODO for having such a page